context/index.d.ts

```diff
@@ -0,0 +1,17 @@
+import type { ReadableAtom } from '../atom'
+
+// This isn't a part of the public API for now
+type Context = unknown
+
+export type ContextGetter = <T extends ReadableAtom>(store: T) => T
+export type WithCtx = {
+  ctx: ContextGetter
+}
+
+export declare const globalContext: Context
+
+export function createContext(storeValues?: object): Context
+export function resetContext(context?: Context): void
+export function serializeContext(context: Context): string
+
+export function withContext<T>(storeOrAction: T, ctx: Context): T
```

Comment:
I think we can also start writing docs in TSDoc. The public API looks good, but docs will help us to see the bigger picture.